### PR TITLE
[Issue {#25}] Add new dataset for clustering problems inside datasets folder

### PR DESCRIPTION
**Pull request by snega16**
**This PR fixes for issue [#25](https://github.com/sagnik1511/Tabular-AutoML/issues/25)**

**Changes made:**
 - Added new clustering dataset Credit Card Customer data to datasets 
 - Updated datasets,py with clustering class clustering() and clustering dataset class called Credit_Card_Customer_Data().

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -61,12 +61,12 @@ def regression_test():
     trainer.single_model_trainer(x_train=x_train, y_train=y_train, x_val=x_val, y_val=y_val, save_model=True)
     print("Regression test completed successfully...\n")
 
-
 def test():
     # Testing classification
     classification_test()
     # Testing Regression
     regression_test()
+
```
Comment (sagnik1511) [inline test.py:70]: Delete this space line.

File: test.py [hunk 2]
```diff
@@ -61,12 +61,12 @@ def regression_test():
     trainer.single_model_trainer(x_train=x_train, y_train=y_train, x_val=x_val, y_val=y_val, save_model=True)
     print("Regression test completed successfully...\n")
 
-
 def test():
```
Comment (sagnik1511) [inline test.py]: add the space line as added before.

File: tab_automl/automl/datasets.py [hunk 3]
```diff
@@ -77,6 +77,35 @@ def prepare_x_and_y(self,
         print("X feature set and target feature has been split...")
 
         return x, y
+
```
Comment (sagnik1511) [inline tab_automl/automl/datasets.py:80]: Always add 2 lines of space before adding a new class. In this case, add another newline here.

File: tab_automl/automl/datasets.py [hunk 4]
```diff
@@ -161,3 +190,37 @@ def prepare_x_and_y(self,
...
+    def __init__(self):
+        # reading the dataset from source
+        self.data = pd.read_csv("tab_automl/datasets/Credit_Card_Customer_Data.csv")
+        # storing the columns overview
+        self.columns = pd.Series([str(self.data[feature].dtype) for feature in self.data.columns])
+        self.labels = None
+        print(f"Populated the dataframe with data records...")
+
+    def prepare_x(self,
```
Comment (sagnik1511) [inline tab_automl/automl/datasets.py]: In clustering, as all the columns will be used for clustering, there's no need to add `feature_set_columns`.

**Comment by snega16** [comment]
All changes were made!

**Review by sagnik1511 — APPROVED** [review]

**Pull request merged**